ui-admin/src/study/notifications/EmailTemplateEditor.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useRef, useState } from 'react'
+import EmailEditor, { EditorRef, EmailEditorProps } from 'react-email-editor'
+import { EmailTemplate } from '@juniper/ui-core'
+import { Tab, Tabs } from 'react-bootstrap'
+import { getImageBaseUrl } from 'api/api'
+
+/** Enables editing an email with design/preview modes */
+export default function EmailTemplateEditor({ emailTemplate, updateEmailTemplate, portalShortcode }: {
+    emailTemplate: EmailTemplate, portalShortcode: string, updateEmailTemplate: (emailTemplate: EmailTemplate) => void
+}) {
+  const emailEditorRef = useRef<EditorRef>(null)
+  const [activeTab, setActiveTab] = useState<string | null>('designer')
+
+  const replacePlaceholders = (html: string) => {
+    return html.replaceAll('${siteImageBaseUrl}', location.origin + getImageBaseUrl(portalShortcode))
+  }
+  const insertPlaceholders = (html: string) => {
+    return html.replaceAll(location.origin + getImageBaseUrl(portalShortcode), '${siteImageBaseUrl}')
+  }
+
+  const onEditorLoaded: EmailEditorProps['onReady'] = unlayer => {
+    unlayer.loadDesign({
+      // @ts-ignore
+      html: replacePlaceholders(emailTemplate.body),
+      classic: true
+    })
+    unlayer.addEventListener('design:updated', () => {
+      if (!emailEditorRef.current?.editor) { return }
+      emailEditorRef.current.editor.exportHtml(data => {
+        updateEmailTemplate({
+          ...emailTemplate,
+          body: insertPlaceholders(data.html)
+        })
+      })
+    })
+  }
+
+  const templateVersionString = `v${emailTemplate.version}`
+  return <div className="mt-3">
+    <div className="d-flex align-items-baseline">
+      <h3 className="h6">{emailTemplate.name}</h3>
+      <div className="ms-2 text-muted fst-italic">
+                ({emailTemplate.stableId} {templateVersionString})
+      </div>
+    </div>
+    <div>
+      <label className="form-label">Subject
+        <input className="form-control" type="text" size={100} value={emailTemplate.subject}
+          onChange={e => updateEmailTemplate({
+            ...emailTemplate,
+            subject: e.target.value
+          })}/>
+      </label>
+    </div>
+    <div>
+      <Tabs
+        activeKey={activeTab ?? undefined}
+        className="mb-1"
+        mountOnEnter
+        unmountOnExit
+        onSelect={setActiveTab}
+      >
+        <Tab eventKey="designer" title="Designer">
+          <EmailEditor
```

Comment:
Testing note: I get `GET https://editor.unlayer.com/1.5.33/tinymce/skins/content/default/content.css net::ERR_ABORTED 403 (Forbidden)` when the editor first loads. 

I also get an error in the console when I try to collapse the toolbar:

`editor.js:2 Uncaught TypeError: Cannot read properties of undefined (reading 'style')
    at n (editor.js:2:709842)
    at Object.onAction (editor.js:2:711331)
    at editor.js:2:5489116
    at editor.js:2:5297770
    at editor.js:2:5489077
    at editor.js:2:5196794
    at V (editor.js:2:5163973)
    at run (editor.js:2:5196771)
    at editor.js:2:5197545
    at editor.js:2:5161687`

Reply:
hmm... I noticed that a lot of stuff in the editor was  wonky -- e.g. I spent 20mins trying to futz with options to make the preview read-only, or hide the toolbars, and nothing worked.  I wonder if we'll need to switch to a different, better-supported editor when we revisit this page

Reply:
Yeah, and overall it works fine for now, so definitely not blocking.